client/App.jsx

```diff
@@ -80,6 +81,23 @@ class App extends Component {
     this.setState({ drawerShowing: null });
   }
 
+  beginQrScan(imageBufferData) {
+    return new Promise((resolve, reject) => {
```

Comment:
The big idea here is that we can wrap the "worker ness" with a promise, and the rest of the app is none the wiser.